Realm/ObjectStore/shared_realm.cpp

```diff
@@ -0,0 +1,446 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2015 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#include "shared_realm.hpp"
+
+#include "external_commit_helper.hpp"
+#include "realm_delegate.hpp"
+#include "schema.hpp"
+#include "transact_log_handler.hpp"
+
+#include <realm/commit_log.hpp>
+#include <realm/group_shared.hpp>
+
+#include <mutex>
+
+using namespace realm;
+
+RealmCache Realm::s_global_cache;
+
+Realm::Config::Config(const Config& c)
```

Comment:
The dynamic schema is now always created (and used for the validation, so it's never wasted work), so for dynamic you just don't pass in a custom schema or call update_schema().